### DISCUSSION TITLE
Changes to service granularity over time?

### DISCUSSION THREAD

**Discussion by famkedriessen** [magda-io/magda#3434] [Q&A]
Hi all! I am doing a research in which I analyze git histories of projects working on microservice-based systems, to see whether such a model can point out services that should be split or merged, based on dependencies in the git logs (if service A is 90% of the time updated on the same day as service B, this might be an indication for tight-coupling). To validate my model I am looking for open-source cases in which either some functionality was extracted into a new service, or in which architectural decay has taken place (a service grew bigger and bigger over time while becoming less and less cohesive). Would someone be able to point out such "fluctuations in granularity" in Magda to me? Thank you in advance!

**Comment by t83714** [comment]
I don't recall too many "fluctuations in granularity" examples in Magda.
The only recent change was to make the previous independent OPA pod run as a sidecar of the auth API.
But that's more for load balance consideration, and the code bases are still kept separately.
